src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -35,14 +35,23 @@ public static bool IsTrackedResource(Schema schema, Dictionary<string, Schema> d
             return false;
         }
 
-        // determine if the operation is xms pageable or returns an object of array type
-        public static bool IsXmsPageableOrArrayResponseOperation(Operation op, ServiceDefinition entity)
+        // determine if an operation is xms pageable operation
+        public static bool IsXmsPageableOperation(Operation op)
         {
             // if xmspageable type, return true
-            if (op.Extensions.GetValue<object>(XmsPageable) != null) return true;
+            if (op.Extensions.GetValue<object>(XmsPageable) != null)
+            {
+                return true;
+            }
+
+            return false;
+        }
 
+        // determine if an operation returns an object of array type
+        public static bool IsArrayResponseOperation(Operation op, ServiceDefinition entity)
+        {
             // if a success response is not defined, we have nothing to check, return false
-            if (op.Responses?.ContainsKey("200") !=true) return false;
+            if (op.Responses?.ContainsKey("200") != true) return false;
```

Comment:
In this case because of the conversion from bool? to bool, I need to use a cast such as !(bool)(op.Responses?.ContainsKey("200")) which looks awkward that using != true. So, leaving it as such